sympy/physics/optics/medium.py

```diff
@@ -69,50 +71,34 @@ class Medium(Symbol):
     """
 
     def __new__(cls, name, permittivity=None, permeability=None, n=None):
-        obj = super().__new__(cls, name)
-        obj._permittivity = sympify(permittivity)
-        obj._permeability = sympify(permeability)
-        obj._n = sympify(n)
+        if not isinstance(name, Str):
+            name = Str(name)
+
         if n is not None:
             if permittivity is not None and permeability is None:
-                obj._permeability = n**2/(c**2*obj._permittivity)
-            if permeability is not None and permittivity is None:
-                obj._permittivity = n**2/(c**2*obj._permeability)
-            if permittivity is not None and permittivity is not None:
-                if abs(n - c*sqrt(obj._permittivity*obj._permeability)) > 1e-6:
-                   raise ValueError("Values are not consistent.")
+                permeability = n**2/(c**2*permittivity)
+                return MediumPP(name, _sympify(permittivity), _sympify(permeability))
+            elif permeability is not None and permittivity is None:
+                permittivity = n**2/(c**2*permeability)
+                return MediumPP(name, _sympify(permittivity), _sympify(permeability))
+            elif permittivity is not None and permittivity is not None:
+                expr = abs(n - c * sqrt(permittivity * permeability))
+                expr = expr.subs({meter: 1, second: 1})
+                if len(expr.free_symbols) == 0 and expr > 1e-6:
```

Comment:
I think that passing all 3 arguments should just always be an error. This code for checking consistency is unreliable and shouldn't be attempted